core/pending.go

```diff
@@ -64,29 +73,65 @@ func (p *Pending) GetTransactionStateDiffs() []*StateDiff {
 	return []*StateDiff{}
 }
 
+func (p *Pending) GetPreLatest() *PreLatest {
+	return nil
+}
+
 func (p *Pending) Variant() PendingDataVariant {
 	return PendingBlockVariant
 }
 
+func (p *Pending) Validate(parent *Header) bool {
+	if parent == nil {
+		return p.Block.ParentHash.Equal(&felt.Zero)
+	}
+
+	return p.Block.ParentHash.Equal(parent.Hash)
+}
+
+type PreLatest Pending
+
 type PreConfirmed struct {
 	Block       *Block
 	StateUpdate *StateUpdate
 	// Node does not fetch unknown classes. but we keep it for sequencer
 	NewClasses            map[felt.Felt]Class
 	TransactionStateDiffs []*StateDiff
 	CandidateTxs          []Transaction
+	// Optional field, exists if pre_confirmed is N+2 when latest is N
+	PreLatest *PreLatest
 }
 
-func NewPreConfirmed(block *Block, stateUpdate *StateUpdate, transactionStateDiffs []*StateDiff, candidateTxs []Transaction) PreConfirmed {
+func NewPreConfirmed(
+	block *Block,
+	stateUpdate *StateUpdate,
+	transactionStateDiffs []*StateDiff,
+	candidateTxs []Transaction,
+) PreConfirmed {
 	return PreConfirmed{
 		Block:                 block,
 		StateUpdate:           stateUpdate,
-		NewClasses:            make(map[felt.Felt]Class),
 		TransactionStateDiffs: transactionStateDiffs,
 		CandidateTxs:          candidateTxs,
 	}
 }
 
+func (p *PreConfirmed) WithNewClasses(newClasses map[felt.Felt]Class) {
+	p.NewClasses = newClasses
+}
+
+func (p *PreConfirmed) WithPreLatest(preLatest *PreLatest) {
+	p.PreLatest = preLatest
+}
```

Comment:
I believe that these setter methods should return `p`, so they can be chained. Eg:

`p.WithNewClasses(...).WithPreLatests(...)`

Even if we don't use it in this case, to keep it consistent with the project style